Upgrade Gateway Module

hacpy opened the pull request.

hacpy committed:
Allow early distribution of awards

 * Remove unused call

 * Apply for awards through council

hacpy committed:
Try origin looks better

hacpy committed:
Add test

hacpy committed:
Fix test and benchmark

hacpy committed:
Trustee election by trustee admin

 * Remove election by multisig

 * Remove auto election by relayer

hacpy committed:
Reorg common module

hacpy committed:
Use council origin for bitcoin module

hacpy committed:
Fix clippy

hacpy committed:
Regenerate weight

 * Use ensure instead of if

 * Add benchmark genesis

 * Fix benchmark

AAweidai reviewed (commented):
- [x] Delete secp256k1_verifier
- [x] Fix use
- [x] Add coming bot
- [x] Fix verify_tx_valid
- [x] Add rpc

hacpy committed:
Add coming bot

hacpy committed:
Check hot and cold

 * Remove verifier

hacpy committed:
Add storage query rpcs

hacpy committed:
Reorg use for xgateway module

AAweidai committed:
Fix some details

 * Delete unused func

 * Reorg use

AAweidai committed:
Bump spec version to 32

hacpy committed:
Fix test

  Since the public key is sorted when generating multi-signature addresses, the final generated address will be different compared to the previous one.
